stdlib/format.mli

```diff
@@ -45,7 +45,9 @@
 
    {b Warning}: Since {{!section:formatter}formatters} contain
    mutable state, it is not thread-safe to use the same formatter on multiple
-   domains in parallel without synchronization.
+   domains in parallel without synchronization. This may result in
+   [Invalid_argument "Format: Unsynchronized access to formatter"] to be raised
```

Comment:
I would not specify the payload exactly, so that we can tweak it later and still complain at users that would have tried to match on it literally. Just [Invalid_argument] should be enough information.

Reply:
Done